images/win/scripts/ImageHelpers/InstallHelpers.ps1

```diff
@@ -161,6 +161,75 @@ Hashtable for service arguments
     }
 }
 
+function Install-VsixExtension
+{
+    Param
+    (
+        [String]$Url,
+        [String]$Name
+    )
+
+    $FilePath = "${env:Temp}\$Name"
+    $ReleaseInPath = 'Enterprise'
+    $exitCode = -1
+    $retries = 20
+
+    while($true)
+    {
+        try
+        {
+            Write-Host "Downloading $Name..."
+            (New-Object System.Net.WebClient).DownloadFile($Url, $FilePath)
+            break
+        }
+        catch
+        {
+            Write-Host "There is an error during $Name downloading"
+            $_
+            if ($retries -gt 0)
+            {
+                $retries--
+                Write-Host "Waiting 30 seconds before retrying. Retries left: $retries"
+                Start-Sleep -Seconds 30
+            }
+            else {
+                Write-Host "File can't be downloaded"
+                $_
+                exit 1
+            }
+        }
+    }
```

Comment:
loop looks a bit extra complex with `while($true)`, may be something like this will also work?
```
while($retries -gt 0)
{
    try
    {
        Write-Host "Downloading $Name..."
        (New-Object System.Net.WebClient).DownloadFile($Url, $FilePath)
        break
    }
    catch
    {
        Write-Host "There is an error during $Name downloading"
        $_

        $retries--

        if ($retries -eq 0) {
            Write-Host "File can't be downloaded"
            $_
            exit 1
        }

        Write-Host "Waiting 30 seconds before retrying. Retries left: $retries"
        Start-Sleep -Seconds 30
    }
}
```

Reply:
That makes sense! Changed, thank you.

Reply:
@AlenaSviridenko Maybe we should remove duplicate error message?
```
 Write-Host "File can't be downloaded"
            $_
```

Reply:
@al-cheb I think `$_` contains information about why it can't be dowloaded, like response error code or something else, so it doesn't looks like a duplication between `$_` and `File can't be downloaded`
please, correct me if I am missing something

Reply:
@AlenaSviridenko:

First occurence:
```
Write-Host "There is an error during $Name downloading"
        $
```

Second occurence:
```
Write-Host "File can't be downloaded"
            $_
```

Reply:
@al-cheb oh, you mean that duplication. Then yes, I agree, we can remove it